cdap-ui/app/cdap/components/PipelineList/DeployedPipelineView/index.tsx

```diff
@@ -70,25 +82,87 @@ const DeployedPipeline: React.FC = () => {
 
-    let errors = graphQLErrors
-      .concat(networkErrors)
-      .map((err) => err.message)
-      .join('\n');
+    const allErrors: IErrorForDisplay[] = graphQLErrors.concat(networkErrors).map((err) => {
+      const stack = objectQuery(err, 'extensions', 'exception', 'stacktrace');
+      const errorOrigin = objectQuery(err, 'extensions', 'exception', 'errorOrigin');
+
+      return {
+        message: err.message,
+        stack,
+        errorOrigin,
+      };
+    });
 
-    if (!errors || errors.length === 0) {
+    if (!allErrors || allErrors.length === 0) {
       const prefix = /^GraphQL error\:/;
-      errors = error.message.replace(prefix, '').trim();
+      errorForDisplay = {
+        message: error.message.replace(prefix, '').trim(),
+        stack: {},
+        errorOrigin: 'generic',
+      };
+    } else {
+      // Categorizing errors by their origin.
+      allErrors.forEach((eachError: IErrorForDisplay) => {
+        if (!eachError.errorOrigin) {
+          eachError.errorOrigin = 'generic';
+        }
+        if (errorsByOrigin.hasOwnProperty(eachError.errorOrigin)) {
+          errorsByOrigin[eachError.errorOrigin].push(eachError);
+        } else {
+          errorsByOrigin[eachError.errorOrigin] = [eachError];
+        }
+      });
+
+      const errorTypes = Object.keys(errorsByOrigin);
+      // Finding the first high priority error for display.
+      const hpError = errorTypes.find((errorType) => HIGH_PRIORITY_ERRORS.has(errorType));
+
+      if (hpError) {
+        // If there is a high priorty error, pick that one.
+        errorForDisplay = errorsByOrigin[hpError][0];
+        hasHighPriorityError = true;
+      } else if (errorTypes.length > 1) {
+        // When more than one service is down i.e different error types, consider it high priority.
+        hasHighPriorityError = true;
+        errorForDisplay = allErrors[0];
+        errorForDisplay.message =
+          'Multiple services ran into issues when trying to retrieve list of pipelines. Please try again later.';
+      } else {
+        // Pick the first error to display to the user.
+        errorForDisplay = allErrors[0];
+      }
     }
 
-    return <div className="pipeline-deployed-view error-container">{errors}</div>;
+    if (hasHighPriorityError) {
+      return (
+        <div className="pipeline-deployed-view">
+          <Page500 message={errorForDisplay.message} stack={errorForDisplay.stack} />
+        </div>
+      );
+    }
   }
 
+  function generateErrorBannerMessage() {
+    const errorTypesCount = Object.keys(errorsByOrigin).length;
+    return errorForDisplay
+      ? `${errorForDisplay.message}${
+          errorTypesCount > 1
+            ? ` and ${errorTypesCount - 1} other error${
+                errorTypesCount - 1 > 1 ? 's' : ''
+              } occured.`
+            : ''
+        }`
+      : 'Error occured while trying to retrieve information required for list of pipelines.';
```

Comment:
I18n? Also, how can we localize the GraphQL messages?

Reply:
Most of the error messages would come from graphQL/backend and they are not localized. Not sure how we can internationalize those, let me know if you have any ideas.

Fixed the constant strings with i18n.